samcli/local/common/runtime_template.py

```diff
@@ -38,7 +38,7 @@
     ],
     "dotnet": [
         {
-            "runtimes": ["dotnetcore3.1", "dotnetcore2.1"],
+            "runtimes": ["dotnet5.0", "dotnetcore3.1", "dotnetcore2.1"],
```

Comment:
you should not need this as these for zip based runtimes, can you confirm this is needed?

Reply:
Yes this is required else template clone step fails with
```
Error: Lambda Runtime dotnet5.0 and dependency manager cli-package does not have an available initialization template.
```

Reply:
We should figure this out deeper. As Sriram said, this is for zips only. There could be something tied together that shouldn't be since this is the first language that we do not have both a zip and image package type for

Reply:
What's the next step here for this PR?